mevolib/cluster/__init__.py

```diff
@@ -1,121 +1,79 @@
-#   VERSION :  v1.00
-#   AUTHOR(s) :  J. Alvarez-Jarreta
-#
-#-------------------------------------------------------------------------------
-
-from __future__ import absolute_import, division
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Functions aimed to provide an easy interface to handle"""
 
 import multiprocessing
 import math
+from typing import List
 
 from Bio import SeqIO
 
-from . import Genes
-from . import NaiveRows
-from . import NaiveCols
-from . import PRD
+from mevolib.cluster import Genes
+from mevolib.cluster import NaiveRows
+from mevolib.cluster import NaiveCols
+from mevolib.cluster import PRD
 
 from mevolib._utils import get_abspath, NUMCORES
-from mevolib._py3k import viewkeys
 
 
-#-------------------------------------------------------------------------------
-
 _METHOD_TO_FUNC = { 'genes': Genes.map_seqs,
                     'rows': NaiveRows.map_seqs,
                     'cols': NaiveCols.map_seqs,
                     'prd': PRD.map_seqs }
 
 
-#-------------------------------------------------------------------------------
-
-def get_tools ( ) :
-    """
-    Returns :
-        list
-            List of clustering methods and software tools included in the
-            current version of MEvoLib.
-    """
-    return ( list(viewkeys(_METHOD_TO_FUNC)) )
-
+def get_tools() -> List:
+    """Returns a list of clustering methods and software tools included in the current version of MEvoLib."""
+    return list(_METHOD_TO_FUNC.keys)
 
 
-def get_subsets ( method, seqfile, fileformat = 'genbank', *args, **kwargs ) :
-    """
-    Division of all the sequences stored in the sequence input file into subsets
-    applying the 'method' function. If 'seqfile' contains a relative path, the
-    current working directory will be used to get the absolute path.
-
-    Arguments :
-        method  ( string )
-            Desired partition method (case-insensitive): genes, naive rows or
-            cols, padded-Recursive-DCM3.
-        seqfile  ( string )
-            Input sequences file.
-        fileformat  ( string ) 
-            Input file format.
-        args & kwargs
-            Non-keyworded and keyworded arguments passed to the selected method.
-            
+def get_subsets(method: str, seqfile: str, fileformat: str = 'genbank', *args: tuple, **kwargs: dict) -> dict:
+    """Division of all the sequences stored in the sequence input file into subsets applying the 'method' 
+    function. If 'seqfile' contains a relative path, the current working directory will be used to get the 
+    absolute path.
+    Returns a dict with the set identifiers as keys and the corresponding sequences as values in lists of 
+    SeqRecord objects.
```

Comment:
```suggestion
    """Division of all the sequences stored in the input file into subsets applying the `method` function.
    
    If `seqfile` contains a relative path, the current working directory will be used to get the absolute path.
```
The "Returns" bit has to go into its own `Returns:` section below.